app/javascript/components/common/Credits.tsx

```diff
@@ -1,38 +1,48 @@
-import React from 'react'
-import { pluralizeWithNumber } from '../../utils'
-import { AvatarGroup, AvatarGroupProps } from '.'
+import React, { useState } from 'react'
+import { pluralizeWithNumber } from '@/utils'
+import { AvatarGroup } from '.'
+import { User } from '../types'
 
 type CreditsProps = {
   topLabel: string
   topCount: number
   bottomLabel?: string
   bottomCount?: number
   fontSize?: number
-} & AvatarGroupProps
+  users: User[]
+  className?: string
+}
 
 export function Credits({
   topLabel,
   topCount,
   bottomLabel,
   bottomCount,
-  max = 2,
-  avatarUrls,
-  fontSize = 14,
+  users,
+  className,
 }: CreditsProps): JSX.Element {
+  const [overflow] = useState<number>(
+    topCount + Number(bottomCount) - users.length
+  )
+
   return (
-    <div
-      className={`flex gap-x-16 text-textColor1 leading-150 whitespace-nowrap text-${fontSize} items-center`}
-    >
-      <AvatarGroup max={max} avatarUrls={avatarUrls} />
-      <div className="flex flex-col gap-y-2">
-        <div className="font-semibold">
-          {pluralizeWithNumber(topCount, topLabel)}
-        </div>
+    <div className={`flex items-center ${className}`}>
+      <AvatarGroup
+        className={`mr-${overflow > 0 ? 12 : 8}`}
+        overflow={overflow}
+        users={users}
+      />
+      <div>
+        {topCount + Number(bottomCount) > 1 || users[0]['handle'].length === 0
```

Comment:
`users[0]['handle'].length === 0` I don't think this is ever true. Every user has a handle. Or are you considering the fact that the data might not be there?

Reply:
yes, just a fallback in case something would happen.